src/include/storage/sql_table.h

```diff
@@ -192,6 +343,39 @@ class SqlTable {
    * @return the projection map for this initializer
    */
   template <class ProjectionInitializerType>
-  ProjectionMap ProjectionMapForInitializer(const ProjectionInitializerType &initializer) const;
+  ProjectionMap ProjectionMapForInitializer(const ProjectionInitializerType &initializer,
+                                            layout_version_t version) const;
+
+  /**
+   * Given a projected row/col translates the column id of each column to the column id of the version passed in
+   * If a column doesn't exist in that version sets the column id to VERSION_POINTER_COLUMN_ID
+   * @param out_buffer - projected row/col whose header to modify
+   * @param curr_dt_version - schema version of the passed in projected row/col
+   * @param old_dt_version - schema version that is desired
```

Comment:
I think the comments are misleading here. These two are not schema versions, but DataTable meta data for that version. (In the above comments you are using 'schema version' to describe version_num instead)